src/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,367 @@
+  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.12), 0px 8px 24px rgba(149, 157, 165, 0.2);
+  min-width: 296px;
+  max-width: min(640px, 100vw);
+  min-height: 480px;
+  max-height: min(640px, 100vh);
+  width: ${props => widthMap[props.width ?? 'auto']};
+  height: ${props => heightMap[props.height ?? 'auto']};
+  border-radius: 12px;
+  overflow: hidden;
+  opacity: 1;
+  animation: overlay--dialog-appear ${ANIMATION_DURATION} ${get('animation.easeOutCubic')};
+
+  @keyframes overlay--dialog-appear {
+    0% {
+      opacity: 0;
+      transform: scale(0.5);
+    }
+    100% {
+      opacity: 1;
+      transform: scale(1);
+    }
+  }
+
+  ${COMMON};
+  ${POSITION};
+  ${sx};
+`
+
+const DefaultHeader: React.FC<DialogHeaderProps> = ({dialogLabelId, title, subtitle, dialogDescriptionId, onClose}) => {
+  const onCloseClick = useCallback(() => {
+    onClose('close-button')
+  }, [onClose])
+  return (
+    <Dialog.Header>
+      <Flex>
+        <Flex pt={2} flexDirection="column" flexGrow={1}>
+          <Dialog.Title id={dialogLabelId}>{title ?? 'Dialog'}</Dialog.Title>
+          {subtitle && <Dialog.Subtitle id={dialogDescriptionId}>{subtitle}</Dialog.Subtitle>}
+        </Flex>
+        <Dialog.CloseButton onClose={onCloseClick} />
+      </Flex>
+    </Dialog.Header>
+  )
+}
+const DefaultBody: React.FC<DialogProps> = ({children}) => {
+  return <Dialog.Body>{children}</Dialog.Body>
+}
+const DefaultFooter: React.FC<DialogProps> = ({footerButtons}) => {
+  const {containerRef: footerRef} = useFocusZone({
+    bindKeys: FocusKeys.ArrowHorizontal | FocusKeys.Tab,
+    focusInStrategy: 'closest'
+  })
+  return footerButtons ? (
+    <Dialog.Footer ref={footerRef as React.RefObject<HTMLDivElement>}>
+      <Dialog.Buttons buttons={footerButtons} />
+    </Dialog.Footer>
+  ) : null
+}
+
+const _Dialog = React.forwardRef<HTMLElement, React.PropsWithChildren<DialogProps>>((props, forwardedRef) => {
+  const {title = 'Dialog', subtitle = '', renderHeader, renderBody, renderFooter, onClose, role = 'dialog'} = props
+  const dialogLabelId = uniqueId()
+  const dialogDescriptionId = uniqueId()
+  const defaultedProps = {...props, title, subtitle, role, dialogLabelId, dialogDescriptionId}
+
+  const dialogRef = useRef<HTMLDivElement>(null)
+  const combinedRef = useCombinedRefs(dialogRef, forwardedRef)
+  const backdropRef = useRef<HTMLDivElement>(null)
+  useFocusTrap({containerRef: dialogRef})
+
+  const callback = useCallback(
+    (event: KeyboardEvent) => {
+      onClose('escape')
+      event.preventDefault()
+    },
+    [onClose]
+  )
+  useOnEscapePress(callback)
+
+  const header = (renderHeader ?? DefaultHeader)(defaultedProps)
+  const body = (renderBody ?? DefaultBody)(defaultedProps)
+  const footer = (renderFooter ?? DefaultFooter)(defaultedProps)
+
+  return (
+    <>
+      <Portal>
+        <Backdrop ref={backdropRef}>
+          <StyledDialog
+            ref={combinedRef as React.RefObject<HTMLDivElement>}
```

Comment:
Can you avoid this cast if you use `HTMLDivElement` in the `React.forwardRef` generic above?